ste/jobStatusManager.go

```diff
@@ -56,19 +61,73 @@ func (jm *jobMgr) SendXferDoneMsg(msg xferDoneMsg) {
 }
 
 func (jm *jobMgr) ListJobSummary() common.ListJobSummaryResponse {
-	jm.jstm.listReq <- true
-	return <-jm.jstm.respChan
+	if !jm.jstm.doneHandleStatusUpdateManager {
+		jm.jstm.listReq <- true
+		return <-jm.jstm.respChan
+	} else {
+		return common.ListJobSummaryResponse{}
+	}
 }
 
 func (jm *jobMgr) ResurrectSummary(js common.ListJobSummaryResponse) {
 	jm.jstm.js = js
 }
 
+// DrainXferDoneMessages() function drain all message on XferDone channel. There is risk of HandleStatusUpdateManager() already exited and we are blocked.
+// We safeguard against it by checking boolean set on exit of HandleStatusUpdateManager(). We try to make risk factor as low as we can, by calling this function only once.
+//
+// Note: This is not thread safe function. Onus on caller to handle that.
+func (jm *jobMgr) DrainXferDoneMessages() bool {
+	if !jm.jstm.doneHandleStatusUpdateManager && !jm.jstm.xferDoneDrainCalled {
+		jm.jstm.xferDoneDrainCalled = true
+		jm.jstm.drainXferDoneSignal <- struct{}{}
+		close(jm.jstm.xferDone)
+		<-jm.jstm.xferDoneDrainedSignal
+		return true
+	} else {
+		jm.Log(pipeline.LogError, fmt.Sprintf("DrainXferDoneMessages already called and its status: %s", common.IffString(jm.jstm.xferDoneDrained, "Success", "Running")))
+		return false
+	}
+}
+
 func (jm *jobMgr) CleanupJobStatusMgr() {
 	jm.Log(pipeline.LogInfo, "CleanJobStatusMgr called.")
 	jm.jstm.done <- struct{}{}
 }
 
+// drainXferDoneCh drains all pending message in xferDone channel.
+func drainXferDoneCh(jm *jobMgr) {
+	jstm := jm.jstm
+	js := &jstm.js
+
+	jm.Log(pipeline.LogError, fmt.Sprintf("len: %v, cap: %v", len(jstm.xferDone), cap(jstm.xferDone)))
+	for msg := range jstm.xferDone {
+		processXferDoneMsg(msg, js)
+	}
+	jstm.xferDoneDrained = true
+	jstm.xferDoneDrainedSignal <- struct{}{}
```

Comment:
nit, comment above this

// Let the sender know that we are done draining all xferdone messages.